src/auto-evo/steps/ForceExtinction.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace AutoEvo
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Godot;
+
+    /// <summary>
+    ///   Forces extinction of worse-faring species to limit the number of steps in next auto-evo run
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     This was a hotfix to the 0.5.6.1 release.
+    ///     TODO: Inquire more refined solution, such as random extinction with bias for smaller populations.
+    ///   </para>
+    /// </remarks>
+    public class ForceExtinction : IRunStep
+    {
+        private readonly List<Patch> patches;
+        private readonly AutoEvoConfiguration configuration;
+
+        public ForceExtinction(List<Patch> patches, AutoEvoConfiguration configuration)
+        {
+            this.patches = patches;
+            this.configuration = configuration;
+        }
+
+        public bool CanRunConcurrently => false;
+
+        public int TotalSteps => 1;
+
+        public bool RunStep(RunResults results)
+        {
+            // We gather populations that may be targeted in the patch,
+            // Depending on the parameters, this may exclude migrated or newly created species.
+            // Excluded species are only protected for one generation, so the player is a target as well,
```

Comment:
Isn't it a lie that the player is a target as the player is skipped anyway on line 70?

Reply:
The line right below precise the player would be rescued if needed. But it is still treated as any other species, otherwise

Reply:
Could you maybe then rename this? As these aren't even the extinction targets, then...

Reply:
I could rename it all as `targetAndPlayer...` but I'm not sure that'd be an improvement in clarity. Really the player is treated as a target for the most part of the computation...

This especially allows to 1) tell the player they are lagging very hard in the competition (as you save them); 2) potentially narrow this rescue to the patch the player currently plays in; 3) enable a potential hardcore mode where the player can go extinct

Reply:
Why is the player even included here, btw?
Even if the player was forced to 0 population by auto-evo, they actually won't even go extinct.
I think I would feel better about this whole variable if the player was excluded.

Reply:
Yeah, but they would be a burden to biodiversity. 

Unlike protected cells who are protected to allow variation in the cellular fauna, the player is protected because it is the player, but they actually don't bring much. So this approach allows the player not to penalize other species whose only wrong would have been to not be protected by the god of entertainment.